debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitMqStreamNativeChangeConsumer.java

```diff
@@ -230,17 +235,41 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitt
                 producer.send(
                         producer.messageBuilder().addData(getBytes(value)).build(),
                         confirmationStatus -> {
+                            try {
+                                if (confirmationStatus.isConfirmed()) {
+                                    committer.markProcessed(record);
+                                }
+                                else {
+                                    LOGGER.error("Failed to confirm message delivery for event '{}'", record);
+                                    hasError.set(true);
+                                }
+                            }
+                            catch (Exception e) {
+                                LOGGER.error("Failed to process record '{}': {}", record, e.getMessage());
+                                hasError.set(true);
+                            }
+                            finally {
+                                latch.countDown();
+                            }
                         });
             }
             catch (StreamException e) {
                 throw new DebeziumException(e);
             }
+        }
 
-            committer.markProcessed(record);
+        if (!latch.await(producerConfirmTimeout, TimeUnit.SECONDS)) {
+            LOGGER.warn("Timeout while waiting for batch confirmation");
+            hasError.set(true);
         }
 
-        LOGGER.trace("Sent messages");
-        committer.markBatchFinished();
+        if (!hasError.get()) {
+            LOGGER.trace("All messages sent successfully");
+            committer.markBatchFinished();
+        }
+        else {
+            LOGGER.error("Batch processing was incomplete due to record processing errors.");
```

Comment:
I believe an exception should be thrown here.

Reply:
done